src/main/java/uk/gov/justice/digital/delius/service/NotificationService.java

```diff
@@ -59,6 +74,22 @@ private void assertAppointmentType(String contactTypeCode) {
         }
     }
 
+    private Optional<Long> matchExistingContact(String crn, NotificationCreateRequest request) {
+        Offender offender = offenderRepository.findByCrn(crn).orElseThrow(() -> new BadRequestException(format("Cannot find offender with CRN: %s", crn)));
+        var contactLocalDate = toLondonLocalDate(request.getContactDateTime());
+        var contactStartLocalTime = toLondonLocalTime(request.getContactDateTime()).truncatedTo(ChronoUnit.SECONDS);
+
+        return contactRepository.findByOffenderAndNsiIdAndContactTypeAndContactDateAndSoftDeletedIsFalse(
```

Comment:
Should this match on the requirement ID as that is a parameter too?

What about the note text/ team / staff member?

Reply:
Will discuss with @mxco86 but the basic reasoning was that within an NSI there should never be a duplicate R&M event for the offender at any point in time.

Reply:
Conclusion is that we are going to make the method CRS specific and in that context the current equality check is adequate. The naming will make it clear.